scripts/download_artifacts.py

```diff
@@ -68,68 +68,92 @@ def get_valid_artifact_hash(
+    # use the same arch extension but the prefix is
+    # currently structured as rvx_zvl_lmulx_ 
+    comps = prefix.split('_')
+    prefix_arch = ''
+
+    if len(comps) == 2:
+        prefix = comps[0]
+    else:
+        prefix = comps[1]
+        prefix_arch = comps[0]
+
+    multilib_arch_extensions = [
+        ext
+        for ext in possible_arch_extensions
+        if prefix in ext
+    ]
+    print("prefix arch:", prefix_arch)
+
+    # now have weekly runners rv32 zvl multilib variants
+    multilib_lists = [
+        "-".join([i, j, "multilib"])
+        for i in libc
+        for j in arch
+    ]
+    if prefix_arch != '':
+        multilib_lists = [name for name in multilib_lists if prefix_arch in name.split('-')[2]]
+
+    multilib_names = [
+        name.format(ext, "{}")
+        for name in multilib_lists
+        for ext in multilib_arch_extensions
+    ]
+
+    return multilib_names
+
+def get_frequent_names() -> List[str]:
     """
-    libc = [f"{prefix}gcc-linux", f"{prefix}gcc-newlib"]
+    Generates all permutaions of target artifact logs for 
+    build frequent runners
+    """
+    libc = [f"gcc-linux", f"gcc-newlib"]
     arch = ["rv32{}-ilp32d-{}", "rv64{}-lp64d-{}"]
 
-    multilib_arch_extensions = None
-    multilib_lists = None
+    multilib_arch_extensions = [
+        "gcv",
+        "imc",
+        "imc_zba_zbb_zbc_zbs",
+    ]
 
-    # Weekly arch extensions included since rv64gcv_zv* doesn't
-    # exist without a prefix
-    if prefix == "":
-        multilib_arch_extensions = [
-            "gcv",
-            "imc",
-            "imc_zba_zbb_zbc_zbs",
-        ]
-
-        multilib_lists = [
-            "-".join([i, j, "multilib"])
-            for i in libc
-            for j in arch
-            if "rv64" in j
-        ]
-    else:
-        possible_arch_extensions = [
-            "gcv_zve64d",
-            "gcv_zvl1024b",
-            "gcv_zvl512b",
-            "gcv_zvl256b",
-        ]
-        # each extension ends in '_'. Use this since lmul extensions
-        # use the same arch extension but the prefix is
-        # currently structured as zvl_lmulx_ 
-        prefix = prefix.split('_')[0]
-        multilib_arch_extensions = [
-            ext
-            for ext in possible_arch_extensions
-            if prefix in ext
-        ]
-
-        # now have weekly runners rv32 zvl multilib variants
-        multilib_lists = [
-            "-".join([i, j, "multilib"])
-            for i in libc
-            for j in arch
-        ]
+    multilib_lists = [
+        "-".join([i, j, "multilib"])
+        for i in libc
+        for j in arch
+        if "rv64" in j
+    ]
 
     multilib_names = [
         name.format(ext, "{}")
         for name in multilib_lists
         for ext in multilib_arch_extensions
+        if not ('linux' in name and 'imc' in ext)
     ]
 
+    remove_names = [name for name in multilib_names if 'linux' in name and 'imc' in name]
```

Comment:
Maybe this is a debug remnant? This will always be blank since the step before filters out these cases

Reply:
Oops missed